h5netcdf/core.py

```diff
@@ -234,8 +245,52 @@ def __init__(self, parent, name):
                     if k.startswith('_nc4_non_coord_'):
                         var_name = k[len('_nc4_non_coord_'):]
                     self._variables.add(var_name)
+
+        if has_existing_unlimited_dims is True:
+            self._determine_current_dimension_sizes()
+
         self._initialized = True
 
+    def _determine_current_dimension_sizes(self):
+        """
+        Helper method to determine the current size of all unlimited
+        dimensions.
+        """
+        def _find_dim(h5group, dim):
+            if dim not in h5group:
+                return _find_dim(h5group.parent, dim)
+            return h5group[dim]
+
+        for d in self.dimensions:
+            if self.dimensions[d] is not None:
+                continue
+            dim = _find_dim(self._h5group, d)
```

Comment:
minor nit: more descriptive variables names for `dim` (`dim_variable`?) and `d` (`dim_name`?) would be appreciated